File: src/serialization/SerializationSimpleTCPServer.java
package serialization;

import java.io.*;
import java.net.*;

class SerializationSimpleTCPServer {
	public static void main(String argv[]) {
		ServerSocket listenSocket;
		try {
			System.out.println("Running server");

			listenSocket = new ServerSocket(6789);
			while (true) {
				Socket connectionSocket = listenSocket.accept();
				ObjectOutputStream outToClient = new ObjectOutputStream(connectionSocket.getOutputStream());
				ObjectInputStream inFromClient = new ObjectInputStream(connectionSocket.getInputStream());
				Pessoa p = (Pessoa) inFromClient.readObject();

				System.out.println("Mensagem do tipo pessoa recebida!");
				System.out.println(p.toString());

				p.cidade = "Fortaleza";
				p.ano = 2014;
				outToClient.writeObject(p);
			}
		} catch (IOException e) {
			e.printStackTrace();
		} catch (ClassNotFoundException e) {
			e.printStackTrace();
		}

	}
}

File: python/serialization/serializationsimpletcpserver.py
import socket, pickle

HOST = 'localhost'
PORT = 6789
s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
s.bind((HOST, PORT))
s.listen(1)
conn, addr = s.accept()

data = conn.recv(4096)
pessoa = pickle.loads(data)

print("Mensagem do tipo pessoa recebida!")
print(pessoa.toString())

pessoa.cidade = "Fortaleza"
pessoa.ano = '2019'
data_string = pickle.dumps(pessoa)
conn.sendall(data_string)
conn.close()